include/fuzzy/ngram_matches.hh

```diff
@@ -1,12 +1,55 @@
 #pragma once
 
 #include <fuzzy/suffix_array.hh>
-#include <fuzzy/agenda_item.hh>
 
 #include <fuzzy/tsl/hopscotch_map.h>
 
 namespace fuzzy
 {
+  // This class tracks the words in the input pattern that are also found in a sentence.
+  class PatternMatch {
+  public:
+    PatternMatch(size_t pattern_length)
+      : _matched_words(pattern_length, false)
+      , _num_matches(0)
+      , _longest_match(0)
+    {
+    }
+
+    // Mark a range of words as matched in a sentence.
+    void set_match(size_t index, size_t length = 1) {
```

Comment:
All these functions should be inline.

Reply:
Methods defined in the class are inline by default:

> A function defined entirely inside a class/struct/union definition, whether it's a member function or a non-member friend function, is implicitly an inline function

https://en.cppreference.com/w/cpp/language/inline